### PR TITLE
Enable TPM in uefi

### PR DESCRIPTION
**Pull request by giggsoff**
To use TPM devices with qemu (when we use uefi provided by eve-uefi package) we should enable support inside uefi. Right now I can see zero values inside PCRs when I try to use eve-uefi.

Also this PR enables TPM option in Makefile to use swtpm as TPM device for Qemu.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

**Review by eriknordmark — COMMENTED** [review]
The qemu parts should be ok, but I don't know the impact of setting the options in pkg/uefi/build.sh - can this have an impact when running on bare metal? @rvs can you take a look?

**Comment by giggsoff** [comment]
@rvs can you please review this? Without this patch PCRs are not filling when we run EVE-OS in qemu using eve-uefi package.

**Comment by mikem-zed** [comment]
I've been using the same path for a wile and I can confirm it is working

File: Makefile [hunk 1]
```diff
@@ -350,39 +360,48 @@ $(EFI_PART) $(BOOT_PART) $(INITRD_IMG) $(INSTALLER_IMG) $(KERNEL_IMG) $(IPXE_IMG
...
 	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive file=$(TARGET_IMG),format=$(IMG_FORMAT)
 
-run-rootfs: $(BIOS_IMG) $(UBOOT_IMG) $(EFI_PART) $(DEVICETREE_DTB)
+run-rootfs: $(BIOS_IMG) $(UBOOT_IMG) $(EFI_PART) $(DEVICETREE_DTB) swtpm
 	(echo 'set devicetree="(hd0,msdos1)/eve.dtb"' ; echo 'set rootfs_root=/dev/vdb' ; echo 'set root=hd1' ; echo 'export rootfs_root' ; echo 'export devicetree' ; echo 'configfile /EFI/BOOT/grub.cfg' ) > $(EFI_PART)/BOOT/grub.cfg
 	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive file=$(ROOTFS_IMG),format=raw -drive file=fat:rw:$(EFI_PART)/..,label=CONFIG,id=uefi-disk,format=vvfat
 	$(QUIET): $@: Succeeded
 
-run-grub: $(BIOS_IMG) $(UBOOT_IMG) $(EFI_PART) $(DEVICETREE_DTB)
+run-grub: $(BIOS_IMG) $(UBOOT_IMG) $(EFI_PART) $(DEVICETREE_DTB) swtpm
```
Comment (rvs) [inline Makefile:404]: Wait, why do we need swtpm in all these targets? I say -- let's make it conditional at least.

**Review by rvs — CHANGES_REQUESTED** [review]
I believe we should be fine when it comes to rebuilding Tianocore with these flags. However, I'm not quite a fan of how it got hooked up to the Makefile logic. See comments.

File: Makefile (hunk 1 above)
Comment (giggsoff) [reply Makefile:404]: Condition is [inside swtpm target](https://github.com/lf-edge/eve/pull/2624/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R367-R370). I want to run swtpm as part of `make run...` command.

**Review by rvs — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -40,6 +40,8 @@ SSH_PROXY=-L6000:localhost:6000
 SSH_KEY=$(CONF_DIR)/ssh.key
 # Use QEMU H/W accelearation (any non-empty value will trigger using it)
 ACCEL=
+# Use tpm (any non-empty value will trigger using it)
+TPM=
```
Comment (eriknordmark) [inline Makefile:44]: Should we document this in the markdown file which has the examples of how to build and run?

**Review by eriknordmark — APPROVED** [review]
Run eden again

**Pull request merged**